### ISSUE TITLE
Google OIDC SSO [Missing scheme in the URI] error

### ISSUE THREAD

**Issue by xmsanchez** [goharbor/harbor#15576]
**Expected behavior and actual behavior:**
When configuring OIDC authentication with Google there's an error in the callback URI.

The expected behaviour is that you are granted with the Google login page.

Actual behaviour is that you receive an 400 HTTP response message and you are not able to log into harbor with this method.

**Steps to reproduce the problem:**
Just setup Google authentication in Harbor.

When you try to login, If you inspect the query parameters, you can see that the redirect_uri indeed does not contain the scheme on it:

![image](https://user-images.githubusercontent.com/48433981/132847953-8cb5784f-c7bd-4301-8c21-a27c4f335ce8.png)

I've double checked in Google Console that the callback URL was well formated (it is, it's mandatory to add the scheme).

I guess the scheme is being trimmed at some point in the source code.

**Versions:**
Please specify the versions of following systems.

- harbor version: 2.1.0
- helm chart version: harbor-1.5.0

**Comment by reasonerjt** [comment]
@xmsanchez 
Did you add `https://` in the redirect_uri in OIDC provider?

**Comment by xmsanchez** [comment]
Yes, in fact this is mandatory, you are not able to apply the configuration if you don't put the scheme. This is the current configuration on Google Cloud console:
![image](https://user-images.githubusercontent.com/48433981/134211179-736a1be0-a425-4112-be4e-2775ce9f1153.png)

**Comment by bitsf** [comment]
you should also paste harbor OIDC configuration UI

**Comment by xmsanchez** [comment]
Sure, sorry:

![image](https://user-images.githubusercontent.com/48433981/134469332-b9288d96-11f5-46e7-8a2e-87ec44a5e4db.png)
